gr-qtgui/python/qtgui/digitalnumbercontrol.py

```diff
@@ -195,7 +197,12 @@ def mousePressEvent(self, event):
                     else:
                         cur_freq -= increment
 
-                    self.setFrequency(cur_freq)
+                    # Cannot call setFrequency to emit.  Change must happen now for
+                    # paint event when clicked.
+                    if (cur_freq >= self.min_freq) and (cur_freq <= self.max_freq):
```

Comment:
I see this block used below verbatim - is it also used elsewhere?  perhaps, wrap into a method

Reply:
Yeah, probably a good idea.  Just pushed update.  I cleaned up the whitespace a little too but left some in for readability.  The control should work like SDR packages where you click on the upper/lower part of the number to change the frequency, so this fixes that functionality.